Mathlib/Algebra/Order/Archimedean/Class.lean

```diff
@@ -452,3 +452,127 @@ theorem liftOrderHom_mk (f : M → α) (h : ∀ a b, mk a ≤ mk b → f a ≤ f
 end LiftHom
 
 end MulArchimedeanClass
+
+variable (M) in
+/-- Subtype of `MulArchimedeanClass` that removes the top element. -/
+@[to_additive ArchimedeanClass₀ "Subtype of `ArchimedeanClass` that removes the top element."]
+abbrev MulArchimedeanClass₁ := {A : MulArchimedeanClass M // A ≠ ⊤}
+
+namespace MulArchimedeanClass₁
+
+/-- Create a `MulArchimedeanClass₁` from a non-one element. -/
+@[to_additive "Create an `ArchimedeanClass₀` from a non-zero element."]
+def mk {a : M} (h : a ≠ 1) : MulArchimedeanClass₁ M :=
```

Comment:
```suggestion
def mk (a : M) (h : a ≠ 1) : MulArchimedeanClass₁ M :=
```
Otherwise you end up with your infoview being full of elided proofs that hold the information about what the context is